src/pages/v2/p/[projectId]/components/V2V3Dashboard.tsx

```diff
@@ -18,15 +19,20 @@ export function V2V3Dashboard({
   return (
     <div style={layouts.maxWidth}>
       <TransactionProvider>
-        <V2V3ProjectMetadataProvider projectId={projectId} metadata={metadata}>
-          <V2V3ProjectProvider projectId={projectId}>
-            <NftRewardsProvider>
-              <VeNftProvider projectId={projectId}>
-                <V2V3Project />
-              </VeNftProvider>
-            </NftRewardsProvider>
-          </V2V3ProjectProvider>
-        </V2V3ProjectMetadataProvider>
+        <V2V3ProjectContractsProvider projectId={projectId}>
```

Comment:
Added this line.

Reply:
Yeah agree that it's become a bit hectic.

I wanted to put the `JBController` contract in the existing V2V3ProjectContext, but some of the hooks in V2V3ProjectProvider need the new `JBController`. So it was either add another parameter to those hooks for JBController, or create a new context. Went with the context option!

As for other contexts, splitting them up gives us a lot of benefits around code re-use. The key is that V1 and V2V3 actually use the same underlying context: `src/contexts/projectMetadataContext.ts`. This means that we can have way more version-agnostic components because they aren't reliant on version-specific contexts.

Definitely open to suggestions on how to improve the nested context hell though lol.

Reply:
What we could do (which actually might be what you're suggesting and I didn't read it right) is keep the existing contexts, but combine them all into a single component.

Lemme see

Reply:
Done this ^. Definitely cleaned it up, and meant that we can re-use it across all the v2v3 pages 🙏 . Thanks for the prompt!!

Reply:
I was more thinking of the contexts, but that all makes sense and squashing the providers is nice 👍